File: 2022-2023/sourcecode/lab/lab13/DaggerHiltBasicsJava/app/src/main/java/pl/edu/uwr/pum/daggerhiltbasicsjava/data/repository/AppRepository.java
package pl.edu.uwr.pum.daggerhiltbasicsjava.data.repository;

import java.util.List;

import pl.edu.uwr.pum.daggerhiltbasicsjava.data.Post;
import pl.edu.uwr.pum.daggerhiltbasicsjava.data.remote.PlaceholderApi;
import retrofit2.Call;

public class AppRepository {
    private final PlaceholderApi api;

    public AppRepository(PlaceholderApi api) {
        this.api = api;
    }

    public Call<List<Post>> getPosts() {
        return api.getPosts();
    }
}

File: 2022-2023/sourcecode/lab/lab13/DaggerHiltBasicsKotlin/app/src/main/java/pl/edu/uwr/pum/daggerhiltbasicskotlin/data/repository/AppRepository.kt
package pl.edu.uwr.pum.daggerhiltbasicskotlin.data.repository

import pl.edu.uwr.pum.daggerhiltbasicskotlin.data.remote.PlaceholderApi
import javax.inject.Inject

class AppRepository @Inject constructor (
    private val api: PlaceholderApi
) {
    suspend fun getPosts() = api.posts()
}